docs/manual/mustcall-checker.tex

```diff
@@ -0,0 +1,164 @@
+\htmlhr
+\chapterAndLabel{Must Call Checker}{mustcall-checker}
+
+The MustCall Checker tracks the methods that an object should call before it is deallocated
+and produces a conservative overapproximation of the set of methods that might
+actually need to be called on an object.
+It is intended to be run as a subchecker of another checker: on its own, it does not
+enforce any rules other than subtyping.
+
+For example, an object whose type is \<java.io.OutputStream> might
+have an obligation to call the \<close()> method, to release an underlying file resource. Or,
+such an \<OutputStream> might not have such an obligation, if the underlying resource is
+a byte array. Either of these obligations can be represented by the static type
+\<@MustCall(\{"close"\}) OutputStream>, which can be read as ``an OutputStream that might need
+to call close before it is deallocated''.
+
+\sectionAndLabel{MustCall annotations}{must-call-annotations}
+
+The MustCall Checker supports these qualifiers and declaration annotations:
```

Comment:
Split the type qualifiers and declaration annotations into two separate lists.
Also, provide example uses -- you will probably want an example for most of the annotations, to help readers understand them more concretely.

Reply:
> Also, provide example uses -- you will probably want an example for most of the annotations, to help readers understand them more concretely

I agree in principle, but doing so without referencing the rest of the algorithm (i.e. the Called Methods and Must Call Consistency checkers) will make the examples strange, which I why I didn't include them. For all of the declaration annotations (`@Owning`, `@NotOwning`, and `@CreatesObligation`) and `@MustCallAlias`, the Must Call Checker implements only a small part of their functionality, and they're difficult to explain without the rest.

For example, `@MustCallAlias` is identical to `@PolyMustCall` for the purposes of the Must Call Checker - only the consistency checker treats them differently.

 I therefore think it's better to just define them and move on, at least for now - but we'll want to make changes and add examples when we upstream the consistency checker, so I'll add appropriate TODOs.